### PR TITLE
feat: use customer id when present whilst creating a premium order

### PR DESCRIPTION
**Pull request by pavelbrm**
### Summary

This PR makes it possible for the known customer to be passed when creating a premium order. Additionally, this makes use of the customer information obtained from the old session when recreating one (mainly as part of the set trial days call).

### Type of Change

- [x] Product feature
- [ ] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before Requesting Review

- [x] Does your code build cleanly without any errors or warnings?
- [ ] Have you used auto closing keywords?
- [x] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [x] Is there a clear title that explains what the PR does?
- [x] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security and/or privacy review if needed
- [ ] Have you performed a self review of this PR?

### CODE REVIEW

**Review by clD11 — APPROVED** [review]
LGTM approved.

File: services/skus/service_nonint_test.go
Comment (kdenhartog) [inline services/skus/service_nonint_test.go:1]: Thanks for adding these tests in here - it made it much simpler to understand what this change was doing.

**Review by kdenhartog — COMMENTED** [review]
LGTM - reviewed this as a part of https://github.com/brave/reviews/issues/1822

No privacy or security concerns

**Comment by pavelbrm** [comment]
> As long as I'm understanding the flow of this properly, I think we are introducing a correlating tracker here that we don't want.

I don't think so. This has always been here in a slightly different form. This data is not kept in SKUs. SKUs used to figure out the customer id itself in some flows, but now it accepts it.

**Pull request merged**